app/Helpers/Fixometer.php

```diff
@@ -220,12 +221,22 @@ public static function userHasEditEventsDevicesPermission($partyId, $userId = nu
         if (is_null($userId)) {
             $userId = Auth::user()->id;
         }
+        $user = User::find($userId);
 
         // Admins can do anything.
-        if (self::hasRole(Auth::user(), 'Administrator')) {
+        if (self::hasRole($user, 'Administrator')) {
             return true;
         }
 
+        if (self::hasRole($user, 'NetworkCoordinator')) {
+            $group = Party::find($partyId)->theGroup;
+            foreach ($group->networks as $network) {
```

Comment:
For reference there's a method $user->isCoordinatorForGroup that does the same check (although differently) - I think your method here is a bit more readable though, and possibly more performant, might be worth updating isCoordinatorForGroup to match.